pkg/frr/cli.go

```diff
@@ -0,0 +1,243 @@
+package frr
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"os/exec"
+	"strconv"
+	"strings"
+
+	"golang.org/x/sys/unix"
+)
+
+type Cli struct {
+	binaryPath string
+}
+
+func NewCli() *Cli {
+	return &Cli{
+		binaryPath: "/usr/bin/vtysh",
+	}
+}
+
+// getVRF returns either the provided vrf from the function params or if its empty.
+// the constant string "all" as the vrf name.
+// if it returns it also provides feedback if the vrf parameter was empty or not as
+// the second return value.
+
+func getVRF(vrf string) (string, bool) {
+	if vrf == "" {
+		return "all", true
+	}
+	return vrf, false
+}
+
+func (frr *Cli) executeWithJSON(args []string) []byte {
+	args = append(args, "json")
+	return frr.execute(args)
+}
+
+func (frr *Cli) execute(args []string) []byte {
+	// Ensure JSON is always appended
+
```

Comment:
```suggestion
```
At least I don't see anything here that would ensure that. Or is that meant as a TODO?